modules/core/src/matrix.cpp

```diff
@@ -1128,7 +1128,7 @@ Mat& Mat::adjustROI( int dtop, int dbottom, int dleft, int dright )
     if(col1 > col2)
         std::swap(col1, col2);
 
-    data += (row1 - ofs.y)*step + (col1 - ofs.x)*esz;
+    data += (row1 - ofs.y)*(int64_t)step + (col1 - ofs.x)*(int64_t)esz;
```

Comment:
I do not think, that it's good for 32bit platforms. M.b. `ptrdiff_t` is better for this purpose?

Reply:
Agreed. I did so so that we can backport to 3.4. I'll go full c++11 then?

Reply:
Yes